addons/web/static/src/js/fields/special_fields.js

```diff
@@ -192,9 +194,71 @@ var FieldReportLayout = relational_fields.FieldMany2One.extend({
 });
 
 
+const IframeWrapper = AbstractField.extend({
+    description: _lt("Wrap raw html within an iframe"),
+
+    // If HTML, don't forget to adjust the sanitize options to avoid stripping most of the metadata
+    supportedFieldTypes: ['text', 'html'],
+
+    template: "web.IframeWrapper",
+
+    _render() {
+
+        const spinner = this.$el.find('.o_iframe_wrapper_spinner')[0];
+        const iframe = this.$el.find('.o_preview_iframe')[0];
```

Comment:
Just so you know, there is a shorthand on widget for this: `this.$('.o_iframe_wrapper_spinner')`. Now if you want to access native elements instead of jQuery I suggest you use `querySelector` on `el` (which is the native version of `$el`) instead, so:
```suggestion
        const spinner = this.el.querySelector('.o_iframe_wrapper_spinner');
        const iframe = this.el.querySelector('.o_preview_iframe');
```